core/src/main/scala/org/http4s/multipart/file/MixedPart.scala

```diff
@@ -0,0 +1,85 @@
+package org.http4s
+package multipart
+package file
+
+import cats.effect.Sync
+import fs2.Stream
+import fs2.io.{readInputStream, file => fs2File}
+import fs2.text.utf8Encode
+import java.io.{File, FileInputStream, InputStream}
+import java.net.URL
+import java.nio.file.{Files, Path}
+
+import org.http4s.headers.`Content-Disposition`
+
+sealed abstract class MixedPart[F[_]] {
```

Comment:
Do we need to introduce a new type here?  Does the consumer get anything out of knowing that the part has been persisted to a temp file?  We could make the contract that you have to `drain` the part body to clean up the temp file.